.github/workflows/python-unit-tests.yml

```diff
@@ -0,0 +1,49 @@
+name: Python unit tests
+
+on: [push]
+
+jobs:
+  tests:
+    name: Test against Python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # If the matrix, you must also update after_n_builds in codecov.yml to match the number of times the test suite is run
+        python-version: ["3.12", "3.11"]
```

Comment:
Do we also suuport a fewer older version of python, or is this based off the "2 latest versions" discusions we have been having

Reply:
The lambda uses Python 3.11 at the moment. I thought it would be worth testing newer versions too so we get early headsup if we write something that won't work with them.